// @toolpad/core/node/DashboardLayout/ThemeSwitcher.js
"use strict";
'use client';

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ThemeSwitcher = ThemeSwitcher;
var React = _interopRequireWildcard(require("react"));
var _material = require("@mui/material");
var _IconButton = _interopRequireDefault(require("@mui/material/IconButton"));
var _Tooltip = _interopRequireDefault(require("@mui/material/Tooltip"));
var _DarkMode = _interopRequireDefault(require("@mui/icons-material/DarkMode"));
var _LightMode = _interopRequireDefault(require("@mui/icons-material/LightMode"));
var _useSsr = _interopRequireDefault(require("@toolpad/utils/hooks/useSsr"));
var _context = require("../shared/context");
var _jsxRuntime = require("react/jsx-runtime");
var _DarkModeIcon, _LightModeIcon;
/**
 * @ignore - internal component.
 */
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function ThemeSwitcher() {
  var _theme$vars;
  const isSsr = (0, _useSsr.default)();
  const theme = (0, _material.useTheme)();
  const {
    paletteMode,
    setPaletteMode,
    isDualTheme
  } = React.useContext(_context.PaletteModeContext);
  const toggleMode = React.useCallback(() => {
    setPaletteMode(paletteMode === 'dark' ? 'light' : 'dark');
  }, [paletteMode, setPaletteMode]);
  return isDualTheme ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_Tooltip.default, {
    title: isSsr ? 'Switch mode' : `${paletteMode === 'dark' ? 'Light' : 'Dark'} mode`,
    enterDelay: 1000,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_IconButton.default, {
        "aria-label": isSsr ? 'Switch theme mode' : `Switch to ${paletteMode === 'dark' ? 'light' : 'dark'} mode`,
        onClick: toggleMode,
        sx: {
          color: ((_theme$vars = theme.vars) != null ? _theme$vars : theme).palette.primary.dark
        },
        children: theme.getColorSchemeSelector ? /*#__PURE__*/(0, _jsxRuntime.jsxs)(React.Fragment, {
          children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_DarkMode.default, {
            sx: {
              display: 'inline',
              [theme.getColorSchemeSelector('dark')]: {
                display: 'none'
              }
            }
          }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_LightMode.default, {
            sx: {
              display: 'none',
              [theme.getColorSchemeSelector('dark')]: {
                display: 'inline'
              }
            }
          })]
        }) : /*#__PURE__*/(0, _jsxRuntime.jsx)(React.Fragment, {
          children: isSsr || paletteMode !== 'dark' ? _DarkModeIcon || (_DarkModeIcon = /*#__PURE__*/(0, _jsxRuntime.jsx)(_DarkMode.default, {})) : _LightModeIcon || (_LightModeIcon = /*#__PURE__*/(0, _jsxRuntime.jsx)(_LightMode.default, {}))
        })
      })
    })
  }) : null;
}
