// @toolpad/core/node/useNotifications/NotificationsProvider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NotificationsProvider = NotificationsProvider;
var React = _interopRequireWildcard(require("react"));
var _material = require("@mui/material");
var _Close = _interopRequireDefault(require("@mui/icons-material/Close"));
var _react2 = require("@toolpad/utils/react");
var _useSlotProps = _interopRequireDefault(require("@mui/utils/useSlotProps"));
var _NotificationsContext = require("./NotificationsContext");
var _jsxRuntime = require("react/jsx-runtime");
var _CloseIcon;
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const closeText = 'Close';
const RootPropsContext = /*#__PURE__*/React.createContext(null);
function Notification({
  notificationKey,
  open,
  message,
  options,
  badge
}) {
  var _props$slots$snackbar, _props$slots, _props$slotProps;
  const {
    close
  } = (0, _react2.useNonNullableContext)(_NotificationsContext.NotificationsContext);
  const {
    severity,
    actionText,
    onAction,
    autoHideDuration
  } = options;
  const handleClose = React.useCallback((event, reason) => {
    if (reason === 'clickaway') {
      return;
    }
    close(notificationKey);
  }, [notificationKey, close]);
  const action = /*#__PURE__*/(0, _jsxRuntime.jsxs)(React.Fragment, {
    children: [onAction ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_material.Button, {
      color: "inherit",
      size: "small",
      onClick: onAction,
      children: actionText != null ? actionText : 'Action'
    }) : null, /*#__PURE__*/(0, _jsxRuntime.jsx)(_material.IconButton, {
      size: "small",
      "aria-label": closeText,
      title: closeText,
      color: "inherit",
      onClick: handleClose,
      children: _CloseIcon || (_CloseIcon = /*#__PURE__*/(0, _jsxRuntime.jsx)(_Close.default, {
        fontSize: "small"
      }))
    })]
  });
  const props = React.useContext(RootPropsContext);
  const SnackbarComponent = (_props$slots$snackbar = props == null || (_props$slots = props.slots) == null ? void 0 : _props$slots.snackbar) != null ? _props$slots$snackbar : _material.Snackbar;
  const snackbarSlotProps = (0, _useSlotProps.default)({
    elementType: SnackbarComponent,
    ownerState: props,
    externalSlotProps: props == null || (_props$slotProps = props.slotProps) == null ? void 0 : _props$slotProps.snackbar,
    additionalProps: {
      open,
      autoHideDuration,
      onClose: handleClose,
      action
    }
  });
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(SnackbarComponent, {
    ...snackbarSlotProps,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_material.Badge, {
      badgeContent: badge,
      color: "primary",
      sx: {
        width: '100%'
      },
      children: severity ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_material.Alert, {
        severity: severity,
        sx: {
          width: '100%'
        },
        action: action,
        children: message
      }) : /*#__PURE__*/(0, _jsxRuntime.jsx)(_material.SnackbarContent, {
        message: message,
        action: action
      })
    })
  }, notificationKey);
}
function Notifications({
  state
}) {
  var _state$queue$;
  const currentNotification = (_state$queue$ = state.queue[0]) != null ? _state$queue$ : null;
  return currentNotification ? /*#__PURE__*/(0, _jsxRuntime.jsx)(Notification, {
    ...currentNotification,
    badge: state.queue.length > 1 ? String(state.queue.length) : null
  }) : null;
}
let nextId = 1;

/**
 * Provider for Notifications. The subtree of this component can use the `useNotifications` hook to
 * access the notifications API. The notifications are shown in the same order they are requested.
 *
 * Demos:
 *
 * - [Sign-in Page](https://mui.com/toolpad/core/react-sign-in-page/)
 * - [useNotifications](https://mui.com/toolpad/core/react-use-notifications/)
 *
 * API:
 *
 * - [NotificationsProvider API](https://mui.com/toolpad/core/api/notifications-provider)
 */
function NotificationsProvider(props) {
  const {
    children
  } = props;
  const [state, setState] = React.useState({
    queue: []
  });
  const show = React.useCallback((message, options = {}) => {
    var _options$key;
    const notificationKey = (_options$key = options.key) != null ? _options$key : `::toolpad-internal::notification::${nextId}`;
    nextId += 1;
    setState(prev => {
      if (prev.queue.some(n => n.notificationKey === notificationKey)) {
        // deduplicate by key
        return prev;
      }
      return {
        ...prev,
        queue: [...prev.queue, {
          message,
          options,
          notificationKey,
          open: true
        }]
      };
    });
    return notificationKey;
  }, []);
  const close = React.useCallback(key => {
    setState(prev => ({
      ...prev,
      queue: prev.queue.filter(n => n.notificationKey !== key)
    }));
  }, []);
  const contextValue = React.useMemo(() => ({
    show,
    close
  }), [show, close]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(RootPropsContext.Provider, {
    value: props,
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)(_NotificationsContext.NotificationsContext.Provider, {
      value: contextValue,
      children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(Notifications, {
        state: state
      })]
    })
  });
}
